cpp/src/slice2cs/Gen.cpp

```diff
@@ -1331,26 +1329,47 @@ Slice::Gen::TypesVisitor::TypesVisitor(IceUtilInternal::Output& out) :
 bool
 Slice::Gen::TypesVisitor::visitModuleStart(const ModulePtr& p)
 {
-    DictionaryList dicts;
-    if(p->hasOnlyDictionaries(dicts))
+    if (p->hasOnlyClassDecls() || p->hasOnlyInterfaces())
```

Comment:
What about the case where a module has both class declarations and interfaces, but both mixed together?
Seems like it might be better to have a method that checks for both in one go, especially since this is the only place where we ever use `hasOnlyClassDecls` and `hasOnlyInterfaces` apparently

Reply:
It's true, but hopefully we're getting rid of these forward-declarations, which will solve the issue.